static/js/components/SiteContentListing.tsx

```diff
@@ -124,16 +121,21 @@ export default function SiteContentListing(): JSX.Element | null {
     type: contenttype,
     offset
   }
-  const [{ isPending: contentListingPending }] = useRequest(
-    websiteContentListingRequest(listingParams)
-  )
+  const [
+    { isPending: contentListingPending },
+    repeatListingRequest
+  ] = useRequest(websiteContentListingRequest(listingParams))
   const listing: WebsiteContentListingResponse = useSelector(
     getWebsiteContentListingCursor
   )(listingParams)
+  const [addConfigItem, setAddConfigItem] = useState<EditableConfigItem | null>(
+    null
+  )
```

Comment:
A bit of a nit, but we can probably change up the names for these vars and the editing equivalents to make them more consistent:

```typescript 
  const [configItemToAdd, setConfigItemToAdd] = useState<EditableConfigItem | null>(
    null
  )
  const [configItemToEdit, setConfigItemToEdit] = useState<{
    uuid: string
    configItem: EditableConfigItem
  } | null>(null)
```

We could also unify these and just set uuid: false for the 'add' case

```typescript
  const [activeConfigItem, setActiveConfigItem] = useState<{
    uuid: string
    configItem: EditableConfigItem
  } | null>(null)
```

Just some ideas. You can pick/choose

Reply:
I like the latter - fewer lines == clearer code imo!

Reply:
Digging into that i found a number of things that seemed redundant or unnecessary, pushing code shortly but basically:

- I made the panel visibility a three-possibility state machine ("none", "edit", "add")
- there doesn't seem to be any need to set the focused `configItem` because when the function is passed down it is, afaict, always called with the `configItem` from the `website`, so I removed this
- then, with just a three-possibility state machine and another piece of state for storing the UUID a few things can be simplified

anyhow, take a look at the most recent commits and lmk what you think